uvicorn/protocols/http.py

```diff
@@ -71,6 +71,8 @@ def put_message(self, message):
     async def receive(self):
         message = await self.receive_queue.get()
         self.protocol.buffer_size -= len(message.get('body', b''))
+        if self.protocol.buffer_size <= 0 and message.get("more_body"):
```

Comment:
I think this check should probably just be `if self.protocol.read_paused`

Reply:
ok, just fix it